### PR TITLE
[9.0][MIG][mass_mailing_custom_unsubscribe] Migrate

### PR DESCRIPTION
**Pull request by yajo**
- Imported last updates from v8.
- Adapted to v9.
- Added a saner default to `mass_mailing.salt` configuration parameter by
  reusing `database.secret` if available, hoping that some day
  odoo/odoo#12040 gets merged.
- Updated README.

@Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please include advices from Olivier in the Odoo PR.

**Comment by pedrobaeza** [comment]
No need to switch to LGPL in this module, as the website part is very minimal.

**Comment by yajo** [comment]
Done, pal :wink:

**Comment by rafaelbn** [comment]
Hi @Yajo , please check travis an runbot, thanks!

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
hmap library needs to be declared. Also a linter error.

**Comment by yajo** [comment]
Yikes, stupid failure: it was `hmac`. Other linters fixed too. I think I should squash last commits before merge.

File: mass_mailing_custom_unsubscribe/migrations/8.0.2.0.0/pre-migrate.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
```
Comment (pedrobaeza) [inline mass_mailing_custom_unsubscribe/migrations/8.0.2.0.0/pre-migrate.py]: Remove this file

Comment (yajo) [reply mass_mailing_custom_unsubscribe/migrations/8.0.2.0.0/pre-migrate.py]: Not, if somebody upgrades to v9 from a version previous than 8.0.2.0.0, they'll need this.

**Comment by pedrobaeza** [comment]
Why the coverage drop if there are tests? Aren't they launched?

File: mass_mailing_custom_unsubscribe/migrations/8.0.2.0.0/pre-migrate.py (hunk 1 above)
Comment (pedrobaeza) [reply mass_mailing_custom_unsubscribe/migrations/8.0.2.0.0/pre-migrate.py]: Before migrating, they should be sure that have updated to latest 8.0 version, or we will need to maintain scripts for this version and the previous one in 2 branches.

**Comment by pedrobaeza** [comment]
I have tried on runbot (configuring a lot of things), and I get a 404 link when clicking on unsubscribing. This is the URL I get: `http://3272307-129-9c55a0.runbot2.odoo-community.org/%3Csmall%3E%3Ca+href=`. Have you tried locally?

**Comment by yajo** [comment]
Maybe I should add some tests...

**Comment by pedrobaeza** [comment]
Yeah, it would save future headaches, and looking for the integration in master, these would be for sure welcome.

**Comment by yajo** [comment]
OK I double-checked v9 implementation and most of this addon must be removed, and the rest refactored. Current required work will be:

- [x] Depend on `website_mass_mailing`
- [x] As such, remove the beautiful unsubscription page, it's bundled.
- [x] Also ~~reuse~~ replace the subscriptions management form they have, to add reasons for unsubscriptions.

**Comment by yajo** [comment]
I hope tests pass...

**Comment by pedrobaeza** [comment]
This is still red

**Comment by pedrobaeza** [comment]
@Yajo, can you finish this?

**Comment by pedrobaeza** [comment]
@Yajo, please finish this

**Comment by yajo** [comment]
Finally tests work in localhost! this should go 💚 now. Not sure which would be the best way to squash here, maybe just squash at merge everything in a single commit by GH interface, since all are mine or transbot's...

**Comment by pedrobaeza** [comment]
I'm afraid still not...

**Comment by pedrobaeza** [comment]
Can you squash previous commits group by author/logical set?

File: mass_mailing_custom_unsubscribe/data/mail_unsubscription_reason.xml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- © 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<openerp>
+<data noupdate="1">
+
+    <record id="reason_not_interested"
+            model="mail.unsubscription.reason"
+            forcecreate="False">
```
Comment (pedrobaeza) [inline mass_mailing_custom_unsubscribe/data/mail_unsubscription_reason.xml:10]: I think you have to put forcecreate="0", as I'm not sure if this is evaluated as `True` as is not an empty string.

Comment (yajo) [reply mass_mailing_custom_unsubscribe/data/mail_unsubscription_reason.xml:10]: Neither is `"0"`. [Check this out.](https://github.com/odoo/odoo/blob/9.0/openerp/tools/convert.py#L211)

Comment (pedrobaeza) [reply mass_mailing_custom_unsubscribe/data/mail_unsubscription_reason.xml:10]: OK, great

**Comment by pedrobaeza** [comment]
@Yajo, what lacks for this PR?

**Comment by yajo** [comment]
Fix the tests (or the code that fails, if tests are OK). Still did not have the time, it's in my TODO list.

**Comment by pedrobaeza** [comment]
Please proceed with the changes so that I can finally merge this.

**Comment by pedrobaeza** [comment]
Still red I'm afraid

**Comment by yajo** [comment]
Bots should go 💚 now.

**Comment by pedrobaeza** [comment]
I'm afraid not.

**Review by pedrobaeza — APPROVED** [review]
Finally green! Make the last squash and merge it

**Comment by yajo** [comment]
Done

**Pull request merged**